payments/schema/mutations.py

```diff
@@ -479,6 +485,79 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteOrderLineMutation()
 
 
+class OrderApprovalInput(graphene.InputObjectType):
+    order_id = graphene.ID(required=True)
+    email = graphene.String(required=True)
+
+
+class ApproveOrderMutation(graphene.ClientIDMutation):
+    class Input:
+        orders = graphene.List(OrderApprovalInput, required=True)
+        due_date = graphene.Date(description="Defaults to the Order due date")
+
+    failed_orders = graphene.List(FailedOrderType, required=True)
+
+    @classmethod
+    @change_permission_required(
+        Order,
+        BerthLease,
+        WinterStorageLease,
+        BerthApplication,
+        WinterStorageApplication,
+    )
+    def mutate_and_get_payload(cls, root, info, **input):
+        from ..notifications import NotificationType
+
+        failed_orders = []
+        due_date = input.get("due_date")
+
+        for order_input in input.get("orders"):
+            order_id = order_input.get("order_id")
+            with transaction.atomic():
+                try:
+                    order = get_node_from_global_id(
+                        info, order_id, only_type=OrderNode, nullable=False,
+                    )
+
+                    # Update due date
+                    if due_date:
```

Comment:
how about defaulting due date to today + 2 weeks? I am thinking that maybe some orders could be generated a while ago before they get sent, so due date originally assigned already passed, even though the order was not yet sent.

Reply:
Makes sense, but in that case, we would never really use the `due date` that is generated with the order, it would always be overwritten by this.

Reply:
true, I was also thinking about that, not sure what is the best approach for this, allowing blank/null dates on the field does not feel like a good solution, so some default is needed.

Reply:
We can follow this approach and just leave the other dates as backup